homeassistant/components/withings/__init__.py

```diff
@@ -213,6 +219,53 @@ async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
     await hass.config_entries.async_reload(entry.entry_id)
 
 
+async def async_subscribe_webhooks(
+    client: ConfigEntryWithingsApi, webhook_url: str
+) -> None:
+    """Subscribe to Withings webhooks."""
+    await async_unsubscribe_webhooks(client)
+
+    notification_to_subscribe = {
+        NotifyAppli.WEIGHT,
+        NotifyAppli.CIRCULATORY,
+        NotifyAppli.ACTIVITY,
+        NotifyAppli.SLEEP,
+        NotifyAppli.BED_IN,
+        NotifyAppli.BED_OUT,
+    }
+
+    for notification in notification_to_subscribe:
+        LOGGER.debug(
+            "Subscribing %s for %s in %s seconds",
+            webhook_url,
+            notification,
+            SUBSCRIBE_DELAY.total_seconds(),
+        )
+        # Withings will HTTP HEAD the callback_url and needs some downtime
+        # between each call or there is a higher chance of failure.
+        await asyncio.sleep(SUBSCRIBE_DELAY.total_seconds())
+        await client.async_notify_subscribe(webhook_url, notification)
```

Comment:
Can the library handle this instead of adding sleeps here?

Reply:
I can move this into my new lib, but all these refractors is required to implement the lib